### PR TITLE
[shopsys] model service layer removal (phase 2)

### PR DESCRIPTION
**Pull request by DavidOstrozlik**
| Q             | A
| ------------- | ---
|Description, reason for the PR| 
|New feature| No 
|BC breaks| Yes 
|Fixes issues|  
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/framework/src/Component/UploadedFile/UploadedFileFactory.php [hunk 1]
```diff
@@ -2,8 +2,24 @@
...
 {
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\FileUpload\FileUpload
+     */
+    private $fileUpload;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\FileUpload\FileUpload $fileUpload
+     */
+    public function __construct(FileUpload $fileUpload)
```
Comment (vitek-rostislav) [inline packages/framework/src/Component/UploadedFile/UploadedFileFactory.php]: should be mentioned in the docs as well...and the `UploadedFileFactoryInterface` should contain the constructor now, right?

File: packages/framework/src/Component/UploadedFile/UploadedFileFactory.php [hunk 2]
```diff
@@ -17,4 +33,24 @@ public function create(
...
         return new UploadedFile($entityName, $entityId, $temporaryFilename);
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\UploadedFile\Config\UploadedFileEntityConfig $uploadedFileEntityConfig
+     * @param int $entityId
+     * @param string[] $temporaryFilenames
+     * @return \Shopsys\FrameworkBundle\Component\UploadedFile\UploadedFile
+     */
+    public function createUploadedFile(
```
Comment (vitek-rostislav) [inline packages/framework/src/Component/UploadedFile/UploadedFileFactory.php]: is it necessary to add the new method? why do not you just move the required logic into `create` method?

Comment (vitek-rostislav) [reply packages/framework/src/Component/UploadedFile/UploadedFileFactory.php]: oh, they require different arguments... :disappointed: the interface is weird now though

Comment (vitek-rostislav) [reply packages/framework/src/Component/UploadedFile/UploadedFileFactory.php]: ...it seems like the `create` method is never called...could it be protected maybe?

File: packages/framework/src/Model/Category/CategoryFactory.php [hunk 3]
```diff
@@ -21,12 +21,18 @@ public function __construct(EntityNameResolver $entityNameResolver)
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Category\CategoryData $data
+     * @param \Shopsys\FrameworkBundle\Model\Category\Category|null $rootCategory
      * @return \Shopsys\FrameworkBundle\Model\Category\Category
      */
-    public function create(CategoryData $data): Category
+    public function create(CategoryData $data, Category $rootCategory = null): Category
```
[the remark this answers is not in the document]
Comment (DavidOstrozlik) [reply packages/framework/src/Model/Category/CategoryFactory.php]: We talked about this and it's possible to have Category::parent null - this happens when you create root category so $rootCategory can be null.

File: packages/framework/src/Model/Category/CategoryFactoryInterface.php [hunk 4]
```diff
@@ -6,7 +6,8 @@
 {
     /**
      * @param \Shopsys\FrameworkBundle\Model\Category\CategoryData $data
+     * @param \Shopsys\FrameworkBundle\Model\Category\Category|null $rootCategory
      * @return \Shopsys\FrameworkBundle\Model\Category\Category
      */
-    public function create(CategoryData $data): Category;
+    public function create(CategoryData $data, Category $rootCategory = null): Category;
```
[the remark this answers is not in the document]
Comment (vitek-rostislav) [reply packages/framework/src/Model/Category/CategoryFactoryInterface.php]: :slightly_smiling_face:

**Comment by vitek-rostislav** [comment]
Hi @DavidOstrozlik - I have added the fixup commit mentioned above, please take a look at it if is ok with you. I see there are many conflicts now so I suggest you to rebase your branch interactively with autosquash, resolve the conflicts and then send the branch for tests :slightly_smiling_face:

**Pull request merged**